### PR TITLE
test refactoring: extract common schema definition from platform checks

### PR DESCRIPTION
**Pull request by nrainer-materialize**

### CODE REVIEW

**Review by def- — COMMENTED** [review]
I'm guilty of having done  this before and I'm wondering if it's a good idea. There is an argument that for tests it might be better to have all of the relevant code in one file so that you can understand the entire test without jumping around. But if I'm the only weirdo without an IDE to make this less of a concern, I'm ok with merging this.

File: misc/python/materialize/checks/common.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+SAMPLE_KAFKA_SCHEMA = """
```
Comment (philip-stoev) [inline misc/python/materialize/checks/common.py:10]: I think `SIMPLE` would be more appropriate in the name.

**Review by philip-stoev — APPROVED** [review]
Thank you!

**Comment by nrainer-materialize** [comment]
> I'm guilty of having done this before and I'm wondering if it's a good idea. There is an argument that for tests it might be better to have all of the relevant code in one file so that you can understand the entire test without jumping around. But if I'm the only weirdo without an IDE to make this less of a concern, I'm ok with merging this.

I don't like the idea of having much duplicated code but I also understand your point of having all you need in front of you.

The options I see here are:
* Keep it as it is.
* Only extract code when the exact details are more or less irrelevant (partially the case here: we only care about receiving some messages from Kafka but don't care about the message structure and content at all; though, statements inserting messages are tied to the structure).
* Only extract code when the exact details are more or less irrelevant and provide functions for some operations: like above with an additional function or parameterized constant to add certain message to the queue to abstract from the structure (maybe overkill)

@philip-stoev, do you have an opinion on this?

File: misc/python/materialize/checks/common.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/common.py:10]: Maybe even better: `KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD` (or similar)

**Comment by philip-stoev** [comment]
I also do not use an IDE and have had the same conflicting thoughts. The Zippy test framework also has the kafka schemas as constants in a file which means they are defined not near the testdrive snippet that will use them , but in a separate python file.

I think the need to put those strings into constants comes from the fact that they are very verbose. If there was a testdrive command that would allow one to define a schema with all important stuff on a single line, in a readable way, then there would be no large json blocks in the tests and there would be no need to put them in a python constant. However, that would be a larger undertaking.

In the absence of such a concise syntax, I would vote for `KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD` . The reason is that if you have an in-line JSON schema, you glance over it and you do not really register if there is any important detail about it or not. But if you give the schema a name, then you sort of know what you are dealing with.

The lack of a library of schemas impacts testing in a subtler way -- one is generally inclined to use `FORMAT BYTES ENVELOPE NONE` to get a shorter test up and running, but this reduces code coverage -- the more interesting, important-for-production and overwhelmingly more bug-prone code paths are with UPSERT or DEBEZIUM, so a schema is required. So our tools should allow people to write UPSERT tests with the same ease as `ENVELOPE NONE`, even if the test does not seem to deal with anything related to envelopes or formats.

In the longer term we have 3 choices I guess:

1. > Only extract code when the exact details are more or less irrelevant and provide functions for some operations: like above with an additional function or parameterized constant to add certain message to the queue to abstract from the structure (maybe overkill)
This is essentially the begginings of a rewrite of testdrive into python. Maybe if we accomodate a lot of misgivings from testdrive and a lot of technical debt due to the use of  the "python + testdrive fragments" paradigm, we will eventually need to go that route.

2. A simpler way to define a schema, something like:

```
$ kafka-define-schema name=schema1 fields=string
```

With even `fields=string` being the default so not having to be mentioned and even the name pre-generated (e.g. defaul to `string-schema` for `fields=string`?


3. Implement a testdrive functionality that will allow you to include some other testdrive file

```
$ include-file=testdrive_file_with_the_json_schema.tdh
```

**Pull request merged**